### PR TITLE
@damassi => Adds basic error page

### PR DESCRIPTION
**Pull request by sweir27**
This PR adds a basic error page to reaction.

Some notes:
- It's just the error page _content_ right now (since it seems that the header is still rendered right now from force? I wasn't totally sure about this. If we need to render a custom header here, hopefully we can do something minimal with just the logo)
- This doesn't hook up the page rendering to anything, it's just the component which accepts `code`, `message`, and `detail`.
- I copied this verbatim from force, but tried to alter the styles within reason to match our design system

image

image

### CODE REVIEW

**Comment by artsyit** [comment]
Deploy preview for *artsy-reaction* ready!

Built with commit aa4686e91d747c716e1fff7b2c6f57d284cdd142

https://deploy-preview-1130--artsy-reaction.netlify.com

File: src/Components/ErrorPage.tsx [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+    const defaultMessage =
+      code === 404
+        ? "Sorry, the page you were looking for doesn’t exist at this URL."
+        : "Internal Error"
+
+    const detailMessage = message ? `Error Message: ${message}` : detail
+
+    return (
+      <>
+        <StyledErrorCodeBackground code={code}>
```
Comment (damassi) [inline src/Components/ErrorPage.tsx]: I think these components can be simplfied to `ErrorCodeBackground`, `ErrorDefaultMessage`, and so on. The idiom for using a `Style` prefix is when needing to wrap (override) an existing component; e.g., 

```js
import styled from 'styled-components'
import { SomeComponent } from 'path/to/SomeComponent'

const StyledSomeComponent = styled(SomeComponent)`
  ...
`

<StyledSomeComponent />
```

File: src/Components/ErrorPage.tsx [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+  max-width: 75%;
+  font-family: "Menlo", "Monaco", "Andale Mono", "lucida console", "Courier New",
+    "monospace";
+  overflow-x: auto;
+  padding: 5px 9px;
+  text-align: left;
+  word-break: break-word;
+`
+
+const StyledErrorCodeBackground = styled<ErrorCodeBackgroundProps, any>("div")`
```
Comment (damassi) [inline src/Components/ErrorPage.tsx]: ooo I didn't realize this TS syntax was possible. Much nicer than previous way

Comment (sweir27) [reply src/Components/ErrorPage.tsx]: I found it somewhere towards the bottom of this issue: https://github.com/styled-components/styled-components/issues/630 (always nice to see friendly faces when reading through these 😄 cc @alloy )

File: src/Components/ErrorPage.tsx (hunk 1 above)
Comment (sweir27) [reply src/Components/ErrorPage.tsx]: Oh! I can get behind that (and wasn't sure about the convention).

**Comment by yuki24** [comment]
This has been deployed to staging by https://github.com/artsy/force/pull/2800

**Pull request merged**